### PR TITLE
Start organizing the change log

### PR DESCRIPTION
**Pull request by strugee**
Incomplete; do not merge. Just wanted to open this to make sure this is something that would get accepted before I put in a lot of work reorganizing everything.

This is based on the recommendations in http://keepachangelog.com/en/0.3.0/

Ref https://github.com/certbot/certbot/pull/4252#issuecomment-283232028

### CODE REVIEW

**Comment by strugee** [comment]
Also, does Certbot adhere to Semantic Versioning?

**Comment by zjs** [comment]
This seems like a useful pattern to adopt.

**Comment by SwartzCr** [comment]
I don't understand why travis doesn't pass for this, I've kicked it so hopefully it'll resolve
@strugee - we claim to support semantic versioning! We should!
I like this PR - I know you mentioned it originally when we first made the change log, let me know how I can help you in this effort to update the format

File: CHANGELOG.md [hunk 1]
```diff
@@ -175,20 +211,23 @@ included on the command line.
...
+* Preserve a certificate's common name when using `renew`
 * Save webroot values for renewal when they are entered interactively
-* Problems with an invalid user-agent string on OS X
-* Gracefully reporting the Apache plugin isn't usable when Augeas is not installed
+* Gracefully report the Apache plugin isn't usable when Augeas is not installed
 * Experimental support for Mageia has been added to `certbot-auto`
 
+### Fixed
+
+* Fix problems with an invalid user-agent string on OS X
```
Comment (strugee) [inline CHANGELOG.md]: This is the one line so far that wasn't clear to me. This was meant to say that a bug was fixed, right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -198,21 +239,24 @@ change the e-mail address associated with your registration.
...
-* `--must-staple` to request certificates from Let's Encrypt with the
-OCSP must staple extension
-* automatic configuration of OSCP stapling for Apache
-* requesting certificates for domains found in the common name of a
+### Added
+
+* Added `--must-staple` to request certificates from Let's Encrypt
+with the OCSP must staple extension
+* Automatically configure OSCP stapling for Apache
+* Allow requesting certificates for domains found in the common name of a
```
Comment (strugee) [inline CHANGELOG.md]: This one's unclear to me too. Does what I changed it to match the intention?

File: CHANGELOG.md [hunk 3]
```diff
@@ -198,76 +244,94 @@ change the e-mail address associated with your registration.
...
-## 05/27/2016
+## 0.7.0 - 2016-05-27
+
+### Added
+
+* Added `--must-staple` to request certificates from Let's Encrypt
+with the OCSP must staple extension
+* Automatically configure OSCP stapling for Apache
+* Allow requesting certificates for domains found in the common name
+of a custom CSR
```
Comment (strugee) [inline CHANGELOG.md]: This one's unclear to me too. Does what I changed it to match the intention?

**Comment by strugee** [comment]
Alrighty, this is done! Review, please :heart:

**Comment by strugee** [comment]
Also, for the record, the change log could use another pass to improve the prose. In particular:

1. Entries feel _very_ verbose; I think a terse summary is probably more useful
2. Verb tenses are inconsistent
3. Some entries are just sentence fragments (although I might've gotten all these? Not sure)
4. There's definitely a couple formatting problems scattered around that I missed

**Comment by strugee** [comment]
Travis failures are (shockingly) unrelated

File: CHANGELOG.md (hunk 1 above)
Comment (SwartzCr) [inline CHANGELOG.md]: I think this is fixing a bug where the macos user agent string was causing an issue with our code. But yes a bug was fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -101,56 +118,80 @@ command either on the command line or in a [configuration
...
-* Adopt more conservative behavior about reporting a needed port as
-unavailable when using the standalone plugin.
 * The Apache plugin uses information about your OS to help determine the
 layout of your Apache configuration directory. We added a patch to
 ensure this code behaves the same way when testing on different systems
 as the tests were failing in some cases.
 
+### Changed
+
+* Adopt more conservative behavior about reporting a needed port as
```
Comment (SwartzCr) [inline CHANGELOG.md]: should this be past tense

**Comment by SwartzCr** [comment]
Looks great - I'm going to assign myself to clean up the prose, but this is a great first pass

File: CHANGELOG.md (hunk 4 above)
Comment (strugee) [inline CHANGELOG.md]: @SwartzCr good question, I dunno... the tenses in general are wildly inconsistent.

**Comment by strugee** [comment]
@SwartzCr thank you!

**Pull request merged**